File: backend/product-service/src/main/java/com/example/productservice/model/Product.java
package com.example.productservice.model;

import jakarta.persistence.*;

import java.time.Instant;

@Entity
@Table(name = "product")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(unique = true)
    private String barcode;

    @Column(nullable = true)
    private String brand;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "generic_product_id")
    private GenericProduct genericProduct;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "image_url")
    private String imageUrl;

    public Product() {
        this.createdAt = Instant.now();
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getBarcode() {
        return barcode;
    }

    public void setBarcode(String barcode) {
        this.barcode = barcode;
    }

    public String getBrand() {
        return brand;
    }

    public void setBrand(String brand) {
        this.brand = brand;
    }

    public GenericProduct getGenericProduct() {
        return genericProduct;
    }

    public void setGenericProduct(GenericProduct genericProduct) {
        this.genericProduct = genericProduct;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }
}

File: frontend/home-inventory-ui/src/types/product.ts
export interface Product {
    id: number;
    name: string;
    brand?: string | null;
    barcode?: string | null;

    createdAt: string;
    imageUrl?: string | null;

    genericProductId?: number | null;
    genericProductName?: string | null;
}

export interface GenericProduct {
    id: number;
    name: string;
    categoryCode: string;
    categoryDisplayName: string;
    imageUrl?: string | null;
}

export interface CreateProductRequest {
    name: string;
    barcode?: string | null;
    brand?: string | null;
    imageUrl?: string | null;

    genericProductId: number;
}